File: src/main/java/shape/Circle.java
package shape;

public class Circle {
    private Point middlePoint;
    private double radiusOne;
    private double radiusTwo;

    public Circle(Point middlePoint, double radiusOne) {
        this.middlePoint = middlePoint;
        this.radiusOne = radiusOne;
        this.radiusTwo = radiusOne;

        if (radiusOne <= 0) {
            throw new IllegalArgumentException("Radius cannot be negative or 0");
        }
    }

    public Point GetMiddlePoint() {
        return middlePoint;
    }
    public double GetRadiusOne() {
        return radiusOne;
    }
    public double GetRadiusTwo() {
        return radiusTwo;
    }
    public double GetArea() {
        double area = Math.PI * radiusOne * radiusTwo;
        return area;
    }

    public void MoveCircle(double addXvalue, double addYvalue) {
        middlePoint.MovePoint(addXvalue, addYvalue);
    }
}

File: src/main/kotlin/shape/Circle.kt
package shape

import kotlin.math.PI

class Circle(
    private val middlePoint: Point,
    private val radius: Double
) {

    init {
        if (radius <= 0) {
            throw IllegalArgumentException("Radius cannot be negative or 0")
        }
    }

    fun GetCenterPoint(): Point {
        return middlePoint
    }

    fun GetRadius(): Double {
        return radius
    }

    fun GetArea(): Double {
        return PI * radius * radius
    }

    fun MoveCircle(addXvalue: Double, addYvalue: Double) {
        middlePoint.MovePoint(addXvalue, addYvalue)
    }
}